package-system/zlib/build_zlib_android.cmd

```diff
@@ -5,14 +5,17 @@
 @rem # SPDX-License-Identifier: Apache-2.0 OR MIT
 @rem #
 
-@IF "%LY_ANDROID_NDK_ROOT%" == "" (
-    @echo Need to set the varible LY_ANDROID_NDK_ROOT to the location of your NDK install!
-    @exit /b 1
-)
 
-cmake -S temp/src -B temp/build -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-fPIC" -DCMAKE_CXX_STANDARD=17 -DCMAKE_TOOLCHAIN_FILE=%LY_ANDROID_NDK_ROOT%\\build\\cmake\\android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DBUILD_SHARED_LIBS=OFF -DSKIP_INSTALL_FILES=YES
+@rem note that toolchain path is relative to the source path (-S) not to the folder this script lives in.
+cmake -S temp/src -B temp/build -G Ninja ^
+    -DCMAKE_BUILD_TYPE=Release ^
+    -DCMAKE_CXX_STANDARD=17 ^
+    -DCMAKE_TOOLCHAIN_FILE=../../../../Scripts/cmake/Platform/Android/Toolchain_android.cmake ^
+    -DBUILD_SHARED_LIBS=OFF ^
+    -DSKIP_INSTALL_FILES=YES
 @if %errorlevel% NEQ 0 ( exit /b 1 )
+
 cmake --build temp/build --target zlibstatic -j 8
 @if %errorlevel% NEQ 0 ( exit /b 1 )
```

Comment:
this builds `Release` only. Are we ok with not making debug versions? 
I am ok (and prefer it) if that is the direction we want to take 3rdparty (that all are compiled in release). For Windows we would have to figure out the iterator debug level flags (and potentially other things), but for the time being we could provide the debug one.

Reply:
Yes, we don't really benefit from debug zlib.  I might even remove it from windows.  We are't in the business of maintaining or debugging zlib.  And the perf of debug is slower than release.  Only reason I provide it in windows is as you say, things like iterator debug level flags.  But in this case, its a pure C library - does that even matter?

Reply:
We only need a separate debug build of a library on Windows, if the library is using the MSVC C++ Standard library and the MSVC C++ Runtime(MSVCRTD)

Reply:
OK, so if its just a staticlib, and its pure-c it doesn't sound like we need to ship debug?

Reply:
I think you can do it for everything except MSVC-platforms (e.g. Windows). Last time I tried, we need all 3rdparty and o3de to set `_ITERATOR_DEBUG_LEVEL` to `0` for all configurations to be able to link debug vs release. AFAIK this also applies to libs.

Reply:
I see.  I should set that as a define.  HOwever, looking at the docs it seems like it only applies to C++, so C-only libraries would not be affected?

Reply:
I think it also applies to C libraries, I think is just a compiler default for `CL`
it can be passed with `-DCMAKE_CXX_FLAGS` 
But I dont think you will be able to do it, for it to work, all 3rdParty needs to either set `_ITERATOR_DEBUG_LEVEL =0` in debug or link against release. And o3de need to also set `_ITERATOR_DEBUG_LEVEL=0` in debug

Reply:
I'm currently testing this to make sure this works on windows debug and release.  CRT DEBUG is no longer active by default in any configuraiton, and iterators only matter if using C++.  ZLIB is compiled using pure C (it doens't even go looking for the c++ compiler or toolchain).

Reply:
It works fine against windows debug